### DISCUSSION TITLE
CharacterSet::unknown use

### DISCUSSION THREAD

**Discussion by priyankasabharwal81** [zxing-cpp/zxing-cpp#750] [Q&A] [answered]
Hello

What is acheived by setting 

ZXing::DecodeHints hints;
hints.setCharacterSet.(CharacterSet::Unknown)

Will it detect the encoding automatically? My application scans barcode images randomly from various sources and its aim is to extract text from these barcode and compare them with my black/whitelist.
I am not sure the use of setting the characterSet to unknown

**Comment by axxel** [comment, marked as answer]
Simply don't touch this property. It should be detected automatically. You can use this if the detection failed and you know the correct character set.
